### PR TITLE
Add support for Nullable Types

### PR DESCRIPTION
**Pull request by vankooch**
Add support for nullable types for the new schema generator. This new PR is a replacement for #992 

Minor change to set nullable option in ```OpenApiSchema``` according to [Data-Types Specs](https://swagger.io/docs/specification/data-models/data-types/). This feature is particularly useful when used in combination with any code generators. I've also added a unit test.

### CODE REVIEW

**Comment by vankooch** [comment]
I've also two other PR in pipeline, any chance we can get this merged in the next days?
One is for updating ReDoc to 2.0.0 RC2 from RC1, the other one is to update the DataTypeAttribute handling in ObjectGenerator, add missing type such as html, uri,...

File: src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGen/SchemaGenerator.cs [hunk 1]
```diff
@@ -38,11 +38,20 @@ public SchemaGenerator(SchemaGeneratorOptions options, JsonSerializerSettings se
 
         public OpenApiSchema GenerateSchema(Type type, SchemaRepository schemaRepository)
         {
+            // Check if is nullable
+            var isNullable = type.IsNullable();
```
Comment (domaindrivendev) [inline src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGen/SchemaGenerator.cs]: Should be `isNullable = type.IsNullable() || type.IsFSharpOption`

**Comment by domaindrivendev** [comment]
One suggested update and fix the conflict then should be good to merge

File: src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGen/SchemaGenerator.cs (hunk 1 above)
Comment (vankooch) [reply src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGen/SchemaGenerator.cs]: Fixed

**Comment by vankooch** [comment]
Should be good to go 😄

**Pull request merged**